feat(graphics/rdr3): add crashometry for vulkan

PappaNiels opened the pull request:
### Goal of this PR
<!-- Concise explanation of what this PR meant to achieve -->
This pr will add crashometry of the resources that are used by the Vulkan API. The data that I added are the following: GPU name, Vulkan APIversion, and driver version. We want to have this data, to diagnose the Vulkan crashes more precisely. To add to that, I also expanded the error handling in glue. Now, instead of having an assert and not having no data, we display the VkResult, so we know what went wrong. 
The validation layers could also be enabled, but it when it is enabled in the instance, it fails and falls back to d3d12. Matter of fact, you need the Vulkan SDK installed, hence the removal for now.

### How is this PR achieving the goal
This pr adds more information to try diagnose crashes, and hopefully guide to a faster solution.


### This PR applies to the following area(s)
<!-- Add any that applies, e.g.: FiveM, RedM, Server, Natives, FxDK, ScRT: Lua, ScRT: C#, ScRT: JS, etc. -->

RedM


### Successfully tested on
<!-- Add any that is applicable, remove any that aren't. -->

**Game builds:** 1491  

**Platforms:** Windows


### Checklist
<!-- Mark all points with x that apply, i.e.: [x]. -->

- [x] Code compiles and has been tested successfully.
- [x] Code explains itself well and/or is documented.
- [x] My commit message explains what the changes do and what they are for.
- [x] No extra compilation warnings are added by these changes.


### Fixes issues
<!-- Add any issue that this PR fixes with: `fixes #123`, `resolves #234`, `closes #345`. -->
NA

Thanks @Ehbw for the patterns.

PappaNiels commented:
Example data that it will spit out in the console:
![image](https://github.com/user-attachments/assets/b0eba32f-f1b5-4a97-8703-76aede6ea4f6)

prikolium-cfx commented on code/components/rage-graphics-rdr3/src/RenderHooks.cpp:
```diff
@@ -0,0 +1,211 @@
+	// the validation layers should be disabled when playing, but enabled when encountering a crash
+	// if we disable these now, we fallback to dx12 for some reason
+#if 0
+	std::vector<const char*> originalLayers(pCreateInfo->enabledLayerCount + 1);
```
Can this be putted under compilation flag or ConVar if that can be ever needed for end-user?

PappaNiels replied:
That was the idea as well. Sadly the validation layers will fail the instance creation as of now. I am debugging what is going on right now,

prikolium-cfx commented on code/components/rage-graphics-rdr3/src/RenderHooks.cpp:
```diff
@@ -0,0 +1,211 @@
+static VkResult __stdcall vkCreateInstanceHook(VkInstanceCreateInfo* pCreateInfo, VkAllocationCallbacks* pAllocator, VkInstance* pInstance)
+{
+	// the validation layers should be disabled when playing, but enabled when encountering a crash
+	// if we disable these now, we fallback to dx12 for some reason
```
What does it mean? I think it will fallback to DirectX12 if VkResult is not VK_SUCCESS, won't it?

PappaNiels replied:
I was a bit tired yesterday I see. What I meant (as mentioned before), the validation layers are disabled because it defaults back to DX12, due to not being able to create an Vulkan instance.

prikolium-cfx commented on code/components/rage-graphics-rdr3/src/RenderHooks.cpp:
```diff
@@ -0,0 +1,211 @@
+	pCreateInfo->ppEnabledLayerNames = originalLayers.data();
+#endif
+
+	static auto _ = ([&physicalDevice]
```
Could you please make it more obvious, like move to external function and use `initialized` flag or so.

PappaNiels replied:
On it! 🫡

PappaNiels commented:
I have updated the pr with the requested changes and I removed the unclear comments. Now, if you set the variable `EnableVulkanValidation` to 1 in the CitizenFX.ini file, you will enable the validation layers. 

The only must is the Vulkan SDK then, which should be documented somewhere where possible, If someone will try to use the validation layers without the SDK, it will mention it in the logs. But a heads up would be better. 

If there are any other concerns, feel free to let me know.

prikolium-cfx commented:
Thank you for the contribution
